src/components/FileBrowser.vue

```diff
@@ -0,0 +1,285 @@
+
+const GIRDER_FOLDER_ENDPOINT = 'folder';
+const GIRDER_ITEM_ENDOINT = 'item';
+const ICON_MAP = {
+  folder: 'folder',
+  item: 'file',
+};
+
+export default {
+  props: {
+    // location should have properties `type`, `id`
+    location: {
+      type: Object,
+      required: true,
+    },
+    selectEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    multiSelectEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    newItemEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    newFolderEnabled: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  mixins: [sizeFormatter],
+  inject: ['girderRest'],
+  data() {
+    return {
+      rowsLoading: false,
+      breadcrumbLoading: false,
+      pagination: {
+        rowsPerPage: 10,
+        page: 1,
+      },
+      selected: [],
+      refreshCounter_: 0, // https://github.com/girder/girder_web_components/issues/39
+    };
+  },
+  asyncComputed: {
+    breadcrumb: {
+      default: { root: {}, path: [] },
+      async get() {
+        this.breadcrumbLoading = true;
+        const breadcrumb = { root: {}, path: [] };
+        let { id, type } = this.location;
+        while (type) {
+          const { data } = (await this.girderRest.get(`${type}/${id}`));
+          const entity = {
+            name: data._modelType !== 'user' ? data.name : data.login,
+            id: data._id,
+            type: data._modelType,
+            parentId: data.parentId,
+            parentType: data.parentCollection,
+          };
+          if (entity.type === 'folder') {
+            breadcrumb.path.unshift(entity);
+            id = entity.parentId;
+            type = entity.parentType;
+          } else {
+            breadcrumb.root = entity;
+            break;
+          }
+        }
+        this.breadcrumbLoading = false;
+        return breadcrumb;
+      },
+    },
+    counts: {
+      default: { nFolders: 0, nItems: 0 },
+      async get() {
+        const endpoint = `${this.location.type}/${this.location.id}/details`;
+        const { data } = (await this.girderRest.get(endpoint));
+        return {
+          nFolders: data.nFolders ? data.nFolders : 0,
+          nItems: data.nItems ? data.nItems : 0,
+        };
+      },
+      watch() {
+        return [
+          this.pagination,
+          this.location,
+        ];
+      },
+    },
+    rows: {
+      default: [],
+      async get() {
+        const request = async (gr, endpoint, params) => {
+          if (params === null) return [];
+          const resp = await gr.get(endpoint, { params });
+          if (resp.status !== 200) return [];
+          return resp.data;
+        };
+        this.rowsLoading = true;
+        const rows = [
+          ...(await request(this.girderRest, GIRDER_FOLDER_ENDPOINT, this.folderParams)),
+          ...(await request(this.girderRest, GIRDER_ITEM_ENDOINT, this.itemParams)),
```

Comment:
I think this is a place we should use Promise.all([]) instead of two await because following `await`s won't be executed until the former ones finish, losing the async concurrency that we should have.